app/v1/controllers/LiabilitiesController.scala

```diff
@@ -70,9 +70,11 @@ class RetrieveLiabilitiesController @Inject()(val authService: EnrolmentsAuthSer
 
   private def errorResult(errorWrapper: ErrorWrapper) = {
     (errorWrapper.error: @unchecked) match {
-      case VrnFormatError | VrnFormatErrorDes | InvalidDateFromError | InvalidDateToError |
-           RuleDateRangeInvalidError | BadRequestError | InvalidDataError => BadRequest(Json.toJson(errorWrapper))
-      case LegacyUnauthorisedError => Forbidden(Json.toJson(errorWrapper))
+      case VrnFormatError | VrnFormatErrorDes | BadRequestError |
+           FinancialDataInvalidDateFromError | InvalidDateFromErrorDes |
+           FinancialDataInvalidDateToError | InvalidDateToErrorDes |
+           FinancialDataInvalidDateRangeError | InvalidDataError
```

Comment:
InvalidDataError is mentioned as BAD_REQUEST - 400... but as per spec it should return as 404

Reply:
Then the spec is wrong. This scenario is already in the ATs as a 400 and it passes against the legacy code.

Reply:
it's correct in the downstream mapping 
`

E4 | 422 | INVALID_DATA | 400 | INVALID_DATA-- | -- | -- | -- | --


`